src/main/scala/de/zalando/beard/filter/FilterManager.scala

```diff
@@ -0,0 +1,70 @@
+package de.zalando.beard.filter
+
+import scala.collection.immutable.Map
+import scalaz._, Scalaz._
+import org.slf4j.LoggerFactory
+
+/**
+  * @author boopathi
+  */
+object FilterManager {
+
+  val logger = LoggerFactory.getLogger(this.getClass())
+
+  type StateRegistry[A] = State[Registry, A]
+
+  val empty: StateRegistry[Unit] = State.state(Registry(Map.empty))
+
+  case class Registry(data: Map[String, Filter]) {
+
+    def get(name: String): Option[Filter] =
+      data.get(name)
+
+    def add(filter: Filter): Registry =
+      Registry(data + (filter.name -> filter))
+
+  }
+
+  def register (filter: Filter): StateRegistry[Unit] = for {
+
+    fDoesNotExist <- State.gets { r: Registry =>
+      r.get(filter.name) match {
+        case Some(f) => false
+        case None => true
+      }
+    }
+
+    _ <- modify[Registry] { r:Registry =>
+      fDoesNotExist ? r.add(filter) | {
+        // TODO: (@boopathi)
```

Comment:
`registry: Registry` was looking too verbose.

`for (var index = 0 ...` vs `for (var i=0 ...`